pkg/zfs/v1alpha1/mount.go

```diff
@@ -0,0 +1,51 @@
+package zfs
+
+import (
+	"github.com/Sirupsen/logrus"
+	"github.com/openebs/zfs-localpv/pkg/agent/v1alpha1"
+	apis "github.com/openebs/zfs-localpv/pkg/apis/openebs.io/core/v1alpha1"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// UnmountAndDetachDisk unmounts the disk from the specified path
+// and logs out of the iSCSI Volume
+func UnmountAndDetachDisk(vol *apis.ZFSVolume, path string) error {
```

Comment:
Do check if it is better to have this as a method of `ZFSVolume` struct.
Try to have different structures for ZFSVolume. One is the API/Schema the other is the one used inside service/business logic.

Reply:
yeah, restructurred the cde.